diff --git a/src/main-executable/main.c b/src/main-executable/main.c
index 69585ac..0bc201a 100644
--- a/src/main-executable/main.c
+++ b/src/main-executable/main.c
@@ -628,7 +628,11 @@ static void open(GtkApplication *app, GFile **files, gint nfiles, const gchar *h
 				g_string_append_printf(desktop_entry, "%s=%s ", envs[i], getenv(envs[i]));
 			}
 		}
-		g_string_append_printf(desktop_entry, "%s ", d->prgname);
+		// the binary path is hardcoded to argv[0], which may not be a
+		// permanent location (e.g. when running from an AppImage), allow
+		// overriding it with an env variable
+		const char *atl_binary_path = getenv("ATL_BINARY_PATH");
+		g_string_append_printf(desktop_entry, "%s ", (atl_binary_path && *atl_binary_path) ? atl_binary_path : d->prgname);
 		g_string_append_printf(desktop_entry, "--gapplication-app-id %s ", package_name);
 		if (d->apk_main_activity_class)
 			g_string_append_printf(desktop_entry, "-l %s ", d->apk_main_activity_class);
